fvm/accountV2Migration/AccountV2Migration.cdc

```diff
@@ -0,0 +1,89 @@
+access(all)
+contract AccountV2Migration {
+
+    access(all)
+    enum StorageFormat: UInt8 {
+        access(all)
+        case Unknown
+
+        access(all)
+        case V1
+
+        access(all)
+        case V2
+    }
+
+    access(all)
+    event Migrated(
+        addressStartIndex: UInt64,
+        count: UInt64
+    )
+
+    access(all)
+    resource Admin {
+        access(all)
+        fun setNextAddressStartIndex(_ nextAddressStartIndex: UInt64) {
+            AccountV2Migration.nextAddressStartIndex = nextAddressStartIndex
+        }
+
+        access(all)
+        fun setBatchSize(_ batchSize: UInt64) {
+            AccountV2Migration.batchSize = batchSize
+        }
+
+        access(all)
+        fun migrateNextBatch() {
+            AccountV2Migration.migrateNextBatch()
+        }
+    }
+
+    access(all)
+    let adminStoragePath: StoragePath
+
+    access(all)
+    var nextAddressStartIndex: UInt64
+
+    access(all)
+    var batchSize: UInt64
+
+    init() {
+        self.adminStoragePath = /storage/accountV2MigrationAdmin
+        self.nextAddressStartIndex = 1
+        self.batchSize = 0
+
+        self.account.storage.save(
+            <-create Admin(),
+            to: self.adminStoragePath
+        )
```

Comment:
This is really small, but the batching logic could probably live in the `Migration` contract instead of in the V2 one

Reply:
See https://github.com/onflow/flow-go/pull/6761#issuecomment-2549713980: The idea is that all logic related to the particular migration, the "account v2 migration", lives in `AccountV2Migration`, and `Migration` only delegates. 

If we add the logic of the "account v2 migration" to the `Migration` contract, we won't be able to reuse it for other purposes, as the contract update rules e.g. don't allow removing enums, etc.